oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtConverter.java

```diff
@@ -16,36 +16,42 @@
 
 package org.springframework.security.oauth2.server.resource.authentication;
 
-import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.core.authority.SimpleGrantedAuthority;
-import org.springframework.security.oauth2.jwt.Jwt;
-import org.springframework.util.StringUtils;
-
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.stream.Collectors;
 
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.util.StringUtils;
+
 /**
  * @author Rob Winch
+ * @author Josh Cummings
  * @since 5.1
  */
-class JwtConverter {
+public class JwtConverter implements Converter<Jwt, AbstractAuthenticationToken> {
```

Comment:
Change `AbstractAuthenticationToken` -> `JwtAuthenticationToken`

Reply:
I actually ran into difficulty with this. When I'm trying to pick this up as a bean, I use `ResolvableType`, e.g. `getBeanNamesForType(ResolvableType)` so that I can differentiate between `Converter`s of other types from the ones that I want. I figured this:

```
context.getBeanNamesForType(ResolvableType.forClassWithGenerics(Converter.class, Jwt.class, AbstractAuthenticationToken.class))
```

would catch both those extending `Conveter<Jwt, AbstractAuthenticationToken>` and `Converter<Jwt, JwtAuthenticationToken>`, but that didn't appear to be the case. It appears to only pick up those Converters with those exact types.

Are you aware of another way I could achieve this?

Reply:
Nevermind, I see your comment below about not performing bean resolution.